From 596534a20299fece06176d79feadb44f156f7d2c Mon Sep 17 00:00:00 2001
From: Krzysztof Adamski <krzysztofx.adamski@intel.com>
Date: Mon, 13 Jun 2016 16:26:58 +0200
Subject: [PATCH 04/11] usb: gadget: f_mtp: fill ss_descriptors in
 function_alloc_mtp_ptp

Following commit:
74ed9ab usb/f_mtp: MTP function's super speed mode is not enabled

Added lines setting ss_descriptors for MTP and PTP in mtp_bind_config
which even then was dead (unsed) function. Becasue of this,
ss_descriptors where never set and device failed to enumerate in
Superspeed mode.

Change-Id: Idc76d2284a6222f6a1dd29dd3567aab45347364b
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-31240
Signed-off-by: Krzysztof Adamski <krzysztofx.adamski@intel.com>
Reviewed-on: https://android.intel.com:443/514967
---
 drivers/usb/gadget/f_mtp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/usb/gadget/f_mtp.c b/drivers/usb/gadget/f_mtp.c
index a105643..72b8f04 100644
--- a/drivers/usb/gadget/f_mtp.c
+++ b/drivers/usb/gadget/f_mtp.c
@@ -1711,9 +1711,11 @@ struct usb_function *function_alloc_mtp_ptp(struct usb_function_instance *fi,
 	if (mtp_config) {
 		dev->function.fs_descriptors = fs_mtp_descs;
 		dev->function.hs_descriptors = hs_mtp_descs;
+		dev->function.ss_descriptors = ss_mtp_descs;
 	} else {
 		dev->function.fs_descriptors = fs_ptp_descs;
 		dev->function.hs_descriptors = hs_ptp_descs;
+		dev->function.ss_descriptors = ss_ptp_descs;
 	}
 	dev->function.bind = mtp_function_bind;
 	dev->function.unbind = mtp_function_unbind;
-- 
1.9.1

